lib/gooddata/lcm/actions/synchronize_clients.rb

```diff
@@ -87,6 +87,19 @@ def call(params)
                    "Details: #{sync_result['links']['details']}")
             end
 
+            if keep_only_previous_masters_count >= 0
+              number_of_deleted_projects = master_projects.count - (keep_only_previous_masters_count + 1)
+
+              if number_of_deleted_projects.positive?
+                begin
+                  removal_master_project_ids = remove_old_master_workspace(params, segment.segment_id, master_projects, number_of_deleted_projects)
+                  remove_release_table(params, domain_name, data_product.data_product_id, segment.segment_id, master_projects, removal_master_project_ids)
```

Comment:
remove_old_master_workspaces (remove multiple workspaces)
remove_release_table --> remove_old_workspaces_from_release_table

Reply:
updated it